### PR TITLE
Only load wp-editor-classic-layout-styles styles for themes without theme.json support

### PR DESCRIPTION
**Pull request by nosolosw**
Part of https://core.trac.wordpress.org/ticket/53175
See https://github.com/WordPress/gutenberg/issues/31322
Related https://github.com/WordPress/gutenberg/pull/30375

## How to test

Activate a theme without `theme.json` support (TwentyTwenty):

- Load the post editor.
- Verify that the `wp-editor-classic-layout-styles-css` external stylesheet is loaded.

Activate a theme with `theme.json` support (TT1-blocks from the github repo not from the WordPress theme's repo, or add a theme.json file to the TwentyTwentyOne theme):

- Load the post editor.
- Verify that the `wp-editor-classic-layout-styles-css` external stylesheet is **not** loaded.

### CODE REVIEW

File: src/wp-includes/script-loader.php [hunk 1]
```diff
@@ -1599,7 +1600,6 @@ function wp_default_styles( $styles ) {
 		'wp-jquery-ui-dialog',
 		// Package styles.
 		'wp-reset-editor-styles',
-		'wp-editor-classic-layout-styles',
```
Comment (youknowriad) [inline src/wp-includes/script-loader.php:1602]: Nit: I think keeping this here would have been fine (register the style but don't add it to the dependencies, so it doesn't load). Maybe some folks could decide to force load it even if they have theme.json.

**Comment by youknowriad** [comment]
This is looking good.

File: src/wp-includes/script-loader.php (hunk 1 above)
Comment (nosolosw) [reply src/wp-includes/script-loader.php]: reverted at https://github.com/WordPress/wordpress-develop/pull/1290/commits/590155bc45ffffb3293d5e0c93a6b723801ffebc

**Comment by youknowriad** [comment]
committed

**Pull request closed without merging**